src/examples/parallel_hdf5_write_dataset.cpp

```diff
@@ -52,10 +52,30 @@ int main(int argc, char** argv) {
 
         // Each node want to write its own rank two time in
         // its associated row
-        int data[1][2] = {{mpi_rank, mpi_rank}};
+        double data[1][2] = {{mpi_rank * 1.0, mpi_rank * 2.0}};
+
+        auto xfer_props = DataTransferProps{};
+        xfer_props.add(UseCollectiveIO{});
 
         // write it to the associated mpi_rank
-        dataset.select({std::size_t(mpi_rank), 0}, {1, 2}).write(data);
+        dataset.select({std::size_t(mpi_rank), 0}, {1, 2}).write(data, xfer_props);
+
+        // Currently, HighFive doesn't wrap retrieving information from property lists.
+        // Therefore, one needs to use HDF5 directly. For example, so see if and why
+        // collective MPI-IO operations were used, one may:
+        uint32_t local_cause = 0, global_cause = 0;
+        auto err = H5Pget_mpio_no_collective_cause(xfer_props.getId(), &local_cause, &global_cause);
+        if (err < 0) {
+            throw std::runtime_error("Failed to check mpio_no_collective_cause.");
+        }
+        if (local_cause || global_cause) {
+            std::cout << "The operation wasn't collective: " << local_cause << " " << global_cause
+                      << std::endl;
+            throw std::runtime_error("IO wasn't collective.");
+        } else {
+            std::cout << "Success! The operation was collective.\n";
```

Comment:
Thanks for adding an example of how to obtain the property list and query it.  In a parallel test, this output will clutter the screen so can we delete the "Success" and emit text only if an error?

Reply:
I know this is an example, and clients can do whatever they want with highfive, but  I think throwing a runtime error is overkill for this condition.  The user requested collective i/o, HDF5 library was unable to perform I/o collectively (for whatever reason), but did perform I/O independently.

Reply:
Good point. The intention behind throwing is that we run the examples a part of our CI. So what you've spotted is that I'm abusing the examples as a test that the request for collective I/O was forwarded correctly to HDF5.

It doesn't cover collective reads, nor all of the entry points. I'll add proper test coverage for this; which frees up the example to be purely instructional. Thank you!